### PR TITLE
Upgrade terraform providers to latest of each used for cluster provision

### PR DESCRIPTION
**Pull request by scottaubrey**
# Description

This change upgrades the providers used by flux clusters to their latest in preparation for https://github.com/elifesciences/issues/issues/7228

Also includes change to remove warning about deprecated properties (`tags` => `tag`) in `aws` provider

### CODE REVIEW

**Comment by scottaubrey** [comment]
Plan output for `flux-test`

    ------------------------------------------------------------------------
    
    An execution plan has been generated and is shown below.
    Resource actions are indicated with the following symbols:
      ~ update in-place
    
    Terraform will perform the following actions:
    
      # aws_autoscaling_group.worker will be updated in-place
      ~ resource "aws_autoscaling_group" "worker" {
            arn                       = "arn:aws:autoscaling:us-east-1:512686554592:autoScalingGroup:0033013d-6760-40d5-b909-1a6a410ecf67:autoScalingGroupName/kubernetes-aws--flux-test--worker"
            availability_zones        = [
                "us-east-1c",
                "us-east-1d",
            ]
            capacity_rebalance        = false
            default_cooldown          = 300
            default_instance_warmup   = 0
            desired_capacity          = 3
            enabled_metrics           = []
            force_delete              = false
          + force_delete_warm_pool    = false
            health_check_grace_period = 300
            health_check_type         = "EC2"
            id                        = "kubernetes-aws--flux-test--worker"
            launch_configuration      = "kubernetes-aws--flux-test--worker20230510080218546200000001"
            load_balancers            = []
            max_instance_lifetime     = 0
            max_size                  = 6
            metrics_granularity       = "1Minute"
            min_size                  = 1
            name                      = "kubernetes-aws--flux-test--worker"
            predicted_capacity        = 0
            protect_from_scale_in     = false
            service_linked_role_arn   = "arn:aws:iam::512686554592:role/aws-service-role/autoscaling.amazonaws.com/AWSServiceRoleForAutoScaling"
            suspended_processes       = []
          ~ tags                      = [
              - {
                  - "key"                 = "Cluster"
                  - "propagate_at_launch" = "true"
                  - "value"               = "kubernetes-aws--flux-test"
                },
              - {
                  - "key"                 = "Environment"
                  - "propagate_at_launch" = "true"
                  - "value"               = "flux-test"
                },
              - {
                  - "key"                 = "Name"
                  - "propagate_at_launch" = "true"
                  - "value"               = "kubernetes-aws--flux-test"
                },
              - {
                  - "key"                 = "Project"
                  - "propagate_at_launch" = "true"
                  - "value"               = "kubernetes-aws"
                },
              - {
                  - "key"                 = "kubernetes.io/cluster/kubernetes-aws--flux-test"
                  - "propagate_at_launch" = "true"
                  - "value"               = "owned"
                },
            ]
            target_group_arns         = []
            termination_policies      = []
            vpc_zone_identifier       = [
                "subnet-0110e7685debdb5eb",
                "subnet-07e912335040a0866",
            ]
            wait_for_capacity_timeout = "10m"
            warm_pool_size            = 0
    
          + tag {
              + key                 = "Cluster"
              + propagate_at_launch = true
              + value               = "kubernetes-aws--flux-test"
            }
          + tag {
              + key                 = "Environment"
              + propagate_at_launch = true
              + value               = "flux-test"
            }
          + tag {
              + key                 = "Name"
              + propagate_at_launch = true
              + value               = "kubernetes-aws--flux-test"
            }
          + tag {
              + key                 = "Project"
              + propagate_at_launch = true
              + value               = "kubernetes-aws"
            }
          + tag {
              + key                 = "kubernetes.io/cluster/kubernetes-aws--flux-test"
              + propagate_at_launch = true
              + value               = "owned"
            }
    
            timeouts {}
        }
    
    Plan: 0 to add, 1 to change, 0 to destroy.
    
    ------------------------------------------------------------------------

These seem fine, and are expected.

**Comment by scottaubrey** [comment]
Plan for `flux-prod`:

```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # aws_autoscaling_group.worker will be updated in-place
  ~ resource "aws_autoscaling_group" "worker" {
        arn                       = "arn:aws:autoscaling:us-east-1:512686554592:autoScalingGroup:5279a493-c78b-4606-a2e7-35630b6410b5:autoScalingGroupName/kubernetes-aws--flux-prod--worker"
        availability_zones        = [
            "us-east-1c",
            "us-east-1d",
        ]
        capacity_rebalance        = false
        default_cooldown          = 300
        default_instance_warmup   = 0
        desired_capacity          = 15
        enabled_metrics           = []
        force_delete              = false
      + force_delete_warm_pool    = false
        health_check_grace_period = 300
        health_check_type         = "EC2"
        id                        = "kubernetes-aws--flux-prod--worker"
        launch_configuration      = "kubernetes-aws--flux-prod--worker20230510082828610200000001"
        load_balancers            = []
        max_instance_lifetime     = 0
        max_size                  = 18
        metrics_granularity       = "1Minute"
        min_size                  = 1
        name                      = "kubernetes-aws--flux-prod--worker"
        predicted_capacity        = 0
        protect_from_scale_in     = false
        service_linked_role_arn   = "arn:aws:iam::512686554592:role/aws-service-role/autoscaling.amazonaws.com/AWSServiceRoleForAutoScaling"
        suspended_processes       = []
      ~ tags                      = [
          - {
              - "key"                 = "Cluster"
              - "propagate_at_launch" = "true"
              - "value"               = "kubernetes-aws--flux-prod"
            },
          - {
              - "key"                 = "Environment"
              - "propagate_at_launch" = "true"
              - "value"               = "flux-prod"
            },
          - {
              - "key"                 = "Name"
              - "propagate_at_launch" = "true"
              - "value"               = "kubernetes-aws--flux-prod"
            },
          - {
              - "key"                 = "Project"
              - "propagate_at_launch" = "true"
              - "value"               = "kubernetes-aws"
            },
          - {
              - "key"                 = "kubernetes.io/cluster/kubernetes-aws--flux-prod"
              - "propagate_at_launch" = "true"
              - "value"               = "owned"
            },
        ]
        target_group_arns         = []
        termination_policies      = []
        vpc_zone_identifier       = [
            "subnet-0110e7685debdb5eb",
            "subnet-07e912335040a0866",
        ]
        wait_for_capacity_timeout = "10m"
        warm_pool_size            = 0

      + tag {
          + key                 = "Cluster"
          + propagate_at_launch = true
          + value               = "kubernetes-aws--flux-prod"
        }
      + tag {
          + key                 = "Environment"
          + propagate_at_launch = true
          + value               = "flux-prod"
        }
      + tag {
          + key                 = "Name"
          + propagate_at_launch = true
          + value               = "kubernetes-aws--flux-prod"
        }
      + tag {
          + key                 = "Project"
          + propagate_at_launch = true
          + value               = "kubernetes-aws"
        }
      + tag {
          + key                 = "kubernetes.io/cluster/kubernetes-aws--flux-prod"
          + propagate_at_launch = true
          + value               = "owned"
        }

        timeouts {}
    }

Plan: 0 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------
```

File: projects/elife.yaml [hunk 1]
```diff
@@ -2050,6 +2050,17 @@ kubernetes-aws:
...
+        version: "0.13.7" # next version is "0.14.11" see `install-terraform.sh`
+        provider-aws:
+            source: "hashicorp/aws"
+            version: "4.66.1"
+        provider-tls:
+            source: "hashicorp/tls"
+            version: "4.0.4"
+        provider-kubernetes:
+            source: "hashicorp/kubernetes"
+            version: "2.20.0"
```
Comment (lsh-0) [inline projects/elife.yaml:2063]: These three providers are only used by `kubernetes-aws` so their global configuration can be changed in the `defaults` section.

I'll do this today.

Comment (lsh-0) [reply projects/elife.yaml:2063]: Done.

**Pull request merged**